frontend/src/profile/Profile.tsx

```diff
@@ -0,0 +1,435 @@
+                        }
+                    </ProfileTitleContainer>
+                    <DataContainer>
+                        <ProfileColumn> 
+                            <ProfileName
+                                isEdit={isEdit}
+                                name={name}
+                                setName={setName}/>
+                            <ProfileMajor
+                                isEdit={isEdit}
+                                major={major.name}
+                                setMajor={setMajor}
+                                majors={majors}/>
+                            <ChangePassword
+                                token={props.token}
+                                id={props.id}/>
+                        </ProfileColumn>
+                        <ProfileColumn> 
+                            <ProfileEmail
+                                isEdit={isEdit}
+                                email={email}
+                                setEmail={setEmail}/>
+                            <ProfileCoop
+                                isEdit={isEdit}
+                                coop={planToString(coop)}
+                                setCoop={setCoop}
+                                plans={plans}
+                                major={major}/>
+                        </ProfileColumn>
+                    </DataContainer>
+                {isEdit && 
+                    <SaveButton 
+                        setEdit={setEdit}
+                        name={name}
+                        email={email}
+                        major={major}
+                        coop={coop}
+                        token={props.token}
+                        id={props.id}
+                        setMajorAction={props.setMajorAction}
+                        setCoopCycleAction={props.setCoopCycleAction}
+                        setFullNameAction={props.setFullNameAction}
+                        setEmailAction={props.setEmailAction}/>
+                }
+                </InnerContainer>
+            </OuterContainer>
+        </Container>
+    )
+}
+
+/**
+ * Callback to be passed into connect, to make properties of the AppState available as this components props.
+ * @param state the AppState
+ */
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+    setMajorAction: (major?: Major) => dispatch(setMajorAction(major)),
+    setCoopCycleAction: (plan: Schedule) => dispatch(setCoopCycle(plan)),
+    setFullNameAction: (fullName: string) => dispatch(setFullNameAction(fullName)),
+    setEmailAction: (email: string) => dispatch(setEmailAction(email))
+});
+  
+/**
+ * Callback to be passed into connect, responsible for dispatching redux actions to update the appstate.
+ * @param dispatch responsible for dispatching actions to the redux store.
+ */
+const mapStateToProps = (state: AppState) => ({
+    name: getFullNameFromState(state),
+    major: getMajorFromState(state),
+    coop: getScheduleFromState(state),
```

Comment:
This was initially done because I needed to schedule to update the state on setCoopCycle in the redux store. But after our discussions regarding multiple schedule, it no longer makes sense to make that update. That should be controlled on a per plan level, so I'll switch this over to use planString